geo/src/algorithm/triangulate_delaunay.rs

```diff
@@ -0,0 +1,642 @@
+    /// for the Delaunay triangle.
+    /// This method uses the determinant of the vertices of the triangle and the
+    /// new point as described by [Guibas & Stolfi](https://doi.org/10.1145%2F282918.282923)
+    /// and on [Wikipedia](https://en.wikipedia.org/wiki/Delaunay_triangulation#Algorithms).
+    pub fn is_in_circumcircle(&self, c: &Coord<T>) -> bool {
+        let a_d_x: f64 = (self.0 .0.x - c.x).into();
+        let a_d_y: f64 = (self.0 .0.y - c.y).into();
+        let b_d_x: f64 = (self.0 .1.x - c.x).into();
+        let b_d_y: f64 = (self.0 .1.y - c.y).into();
+        let c_d_x: f64 = (self.0 .2.x - c.x).into();
+        let c_d_y: f64 = (self.0 .2.y - c.y).into();
+        let a_d_x_d_y = a_d_x.powi(2) + a_d_y.powi(2);
+        let b_d_x_d_y = b_d_x.powi(2) + b_d_y.powi(2);
+        let c_d_x_d_y = c_d_x.powi(2) + c_d_y.powi(2);
+
+        // Compute the determinant of the following matrix
+        // [
+        //     [a_d_x, a_d_y, a_d_x_d_y],
+        //     [b_d_x, b_d_y, b_d_x_d_y],
+        //     [c_d_x, c_d_y, c_d_x_d_y],
+        // ]
+        //
+        let determinant = a_d_x * ((b_d_y * c_d_x_d_y) - (b_d_x_d_y * c_d_y))
+            - a_d_y * ((b_d_x * c_d_x_d_y) - (b_d_x_d_y * c_d_x))
+            + a_d_x_d_y * (b_d_x * c_d_y - b_d_y * c_d_x);
+
+        determinant > 0.0
+    }
+
+    #[cfg(feature = "voronoi")]
+    /// Get the center of the [Circumcircle](https://en.wikipedia.org/wiki/Circumcircle)
+    /// for the Delaunay triangle.
+    pub fn get_circumcircle_center(&self) -> Result<Coord<T>> {
+        // Pin the triangle to the origin to simplify the calculation
+        let b = self.0 .1 - self.0 .0;
+        let c = self.0 .2 - self.0 .0;
+
+        let a_x = self.0 .0.x;
+        let a_y = self.0 .0.y;
+        let b_x = b.x;
+        let b_y = b.y;
+        let c_x = c.x;
+        let c_y = c.y;
+
+        let d = T::from(2.0).ok_or(DelaunayTriangulationError::GeoTypeConversionError)?
+            * (b_x * c_y - b_y * c_x);
+
+        if d.is_zero() {
+            return Err(DelaunayTriangulationError::FailedToComputeCircumcircle);
+        }
+
+        let u_x = (c_y * (b_x.powi(2) + b_y.powi(2)) - b_y * (c_x.powi(2) + c_y.powi(2))) / d;
+        let u_y = (b_x * (c_x.powi(2) + c_y.powi(2)) - c_x * (b_x.powi(2) + b_y.powi(2))) / d;
+
+        Ok(coord! {x: a_x + u_x, y: a_y + u_y})
+    }
+}
+
+/// Delaunay Triangulation Errors
+#[derive(Debug, PartialEq, Eq)]
+pub enum DelaunayTriangulationError {
+    /// Failed to compute the circumcircle for a given triangle.
+    /// This can occur if the points are collinear.
+    FailedToComputeCircumcircle,
+    /// Failed to check if a point is in a circumcircle.
+    FailedToCheckPointInCircumcircle,
+    /// Failed to construct the super triangle.
+    /// This error occurs when the `Polygon` describing the points to
+    /// triangulate does not return a bounding rectangle.
+    FailedToConstructSuperTriangle,
+    FailedToConvertSuperTriangleFactor,
```

Comment:
this and next one need docs

Reply:
I'm going to add the docs as part of the prep for upstream work.